### PR TITLE
Person teacher student classes

### PR DESCRIPTION
**Pull request by V-Blaze**
These are the features implemented in this branch:
- Created a Person class and added its getter and setter functions.
- Created a Student class that inherits the Person class.
- Created a Teacher class that inherits the Person class.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_accessor :name, :age
+
+  attr_reader :id
+
+  def can_use_services?
+    return unless is_of_age? || @parent_permission
```
Comment (gemmen29) [inline person.rb]: - [ ] Kindly fix this typo error as the private function has a name of `of_age?` instead of `is_of_age`.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    true
+  end
```
Comment (gemmen29) [inline person.rb]: - [ ] [OPTIONAL] Kindly change the implementation of the `can_use_services?` to 
```ruby
 def can_use_services? 
    of_age?() || @parent_permission
 end

``` 
as it's easier to read than your implementation and cleaner.

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  private
+
+  def of_age?
+    return true if @age >= 18
+
+    false
+  end
```
Comment (gemmen29) [inline person.rb]: - [ ] [OPTIONAL] Kindly change the implementation of the `of_age?` to 
```ruby
def is_of_age?
    @age >= 18
 end
```
as it's easier to read than your implementation and cleaner.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @V-Blaze, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @V-Blaze, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**